Add automatic tagging and deploy on source index

GiudGiud opened the pull request:
refs #25387 

I think this will be a nice feature of Dempsey, Casey, Jason, Derek and my work on the tagging capabilities of MooseDocs

Figuring out how to create the html page automatically was a little too much for this unfunded pursuit, and unfortunately this precludes having two search pages (there is some more work to do there). So I removed the EM one. The EM one was still a big help in getting this one going so the credit should still go to Casey.

Append modules/filter/index_framework.html on the address of the PR to see the page. I ll post a link when it s up and I get to it

GiudGiud committed:
Add script to add default tags to an existing repo (which wont start from stubs)
refs #25387

GiudGiud committed:
Add automatic tagging capability
refs #25387

GiudGiud committed:
Add filtering pages
refs #25387

GiudGiud committed:
Remove EM tagging as we only support a single tagging page for now

GiudGiud committed:
Partially add capability for supporting more than one filter page
Improve and consolidate file path-based tagging

GiudGiud committed:
Fix NS bib

GiudGiud committed:
Remove my little stunt of duplicating physics pages as both syntax and source pages

GiudGiud committed:
Re-order categories and add the only accurate discretization category

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/2267653/) on 5c260d5 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/27835/site)

This comment will be updated on new commits.

GiudGiud committed:
Final fixups for the example tagging:
- remove a bad tag
- add a missing title
- make background white
- remove tags from EM modules as they get auto-tagged

loganharbour commented:
Not sure how this is automatic?

Nearly all of these can be added automatically by the syntax extension. I don't think requiring another line of boilerplate for every object is a good idea

GiudGiud commented:
I ll look into that extension. 

The boilerplate is needed to trigger the tagging extension only on a selection of pages

GiudGiud commented:
I think we could trigger a new tagging extension on syntax commands. 

That is probably the best option. I ll modify the PR. Thanks for the idea

GiudGiud committed:
Separate tagging expansion away to another command

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/2270435/) on 8854eb7 wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>0445d4</th>
      <th colspan=4>#27835 8854eb</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>85.07%</td>
      <td>85.07%</td>
      <td>-0.00%</td>
      <td>77.39%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>104086</td>
      <td>104085</td>
      <td>-1</td>
      <td>154</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>18264</td>
      <td>18265</td>
      <td>+1</td>
      <td>45</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/27835/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/27835/coverage/framework)


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/27835/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/27835/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/27835/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/27835/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/27835/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/27835/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/27835/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/27835/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/27835/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/27835/coverage/geochemistry)
- [`heat_transfer`](https://mooseframework.inl.gov/docs/PRs/27835/coverage/heat_transfer)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/27835/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/27835/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/27835/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/27835/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/27835/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/27835/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/27835/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/27835/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/27835/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/27835/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/27835/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/27835/coverage/scalar_transport)
- [`solid_mechanics`](https://mooseframework.inl.gov/docs/PRs/27835/coverage/solid_mechanics)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/27835/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/27835/coverage/stochastic_tools)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/27835/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/27835/coverage/xfem)

</details>

## Warnings

- `framework` new line coverage rate 77.39% is less than the suggested 90.0%



This comment will be updated on new commits.
